src/drivers/cmakeDriver.ts

```diff
@@ -390,12 +391,28 @@ export abstract class CMakeDriver implements vscode.Disposable {
         return { vars, variantVars };
     }
 
-    static sourceDirExpansionOptions(workspaceFolderFspath: string | null): expand.ExpansionOptions {
+    static async sourceDirExpansionOptions(workspaceFolderFspath: string | null, kit?: Kit | null, generatorName?: string | null): Promise<expand.ExpansionOptions> {
         const ws_root = util.lightNormalizePath(workspaceFolderFspath || '.');
 
         // Fill in default replacements
+        if (!kit) {
```

Comment:
Same worry here, I'm not sure we want to be using kits every time the cmakeDriver uses this method, since couldn't a preset use this method?